### PR TITLE
CAKE-5511 | Add proxy service

### PR DESCRIPTION
**Pull request by JoshuaRogan**
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5511

## Description
* Route all affiliate links through the redirect service

Instead of sending them to `disney.com/xkdhj39` let's send them to our own service which will redirect them: `https://services.fandom.com/affiliate/redirect/any/data/we/want?r=disney.com/xkdhj39`

## Reviewers

@Wikia/cake

### CODE REVIEW

File: app/utils/affiliate.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const tracking = trackingPairsToObject(unit);
+
+  const category = unit.category;
+
+  // wikiId/articleId/category/algorithm/method/version
+  const path = `${wikiId}/${articleId}/${category}/${tracking.algo}/${tracking.method}/${tracking.version}`;
+
+  const potentialLink = `${host}${path}?r=${encodeURIComponent(link)}`;
+
+  // if we reach the limit lets serve them the link without the tracking
```
Comment (vforge) [inline app/utils/affiliate.js:44]: 👍

Comment (vforge) [reply app/utils/affiliate.js:44]: Can we hit the tracking with some specific value so at least we know how frequently this is triggered?

Comment (JoshuaRogan) [reply app/utils/affiliate.js:44]: Discussed offline and agreed that this event is very unlikely to happen (if ever).

Comment (JoshuaRogan) [reply app/utils/affiliate.js:44]: Discussed offline and agreed that this event is very unlikely to happen (if ever).

**Review by vforge — APPROVED** [review]

File: app/utils/affiliate.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import config from '../config/environment';
+
+function getServiceHost() {
+  if (config.environment === 'production') {
```
Comment (latata) [inline app/utils/affiliate.js:4]: you can use `fetch` service to get the env-specific service url (https://github.com/Wikia/ember-fandom/blob/master/addon/services/fetch.js). you will need to convert this util to the ember service

File: app/utils/affiliate.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    unit.tracking.forEach((kv) => {
+      tracking[`${kv.key}`] = kv.val;
+    });
+  }
+
+  return tracking;
+}
+
+export function linkToProxyLink(link, unit, wikiId, articleId = -1) {
+  // deploy to one community (marvel)
```
Comment (cschucode) [inline app/utils/affiliate.js:29]: I'm curious why marvel needs a link to proxy link?

**Review by vforge — APPROVED** [review]

File: app/utils/affiliate.js (hunk 3 above)
Comment (JoshuaRogan) [reply app/utils/affiliate.js:29]: We are just doing a limited roll out. We will be doing this for every community.

**Pull request merged**